python310Packages.redshift-connector: init at 2.0.910

mcwitt opened the pull request:
###### Description of changes

Adds AWS Redshift Python driver

- https://github.com/aws/amazon-redshift-python-driver
- https://pypi.org/project/redshift-connector/

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mcwitt committed:
python310Packages.redshift-connector: init at 2.0.910

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,58 @@
+
+buildPythonPackage rec {
+  pname = "redshift-connector";
+  version = "2.0.910";
```
```suggestion
  version = "2.0.910";
  format = "setuptools";
```

Set an explicit format for python builds.

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,58 @@
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "amazon-redshift-python-driver";
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

To prevent tag/branch confusion.

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,58 @@
+    scramp
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  checkInputs = [
+    pytest
+    pytest-cov
+    pytest-mock
+  ];
```
```suggestion
  nativeCheckInputs = [
    pytest-mock
    pytestCheckHook
  ];
```

I think all should be `nativeCheckInputs`. We also don't want `pytest-cov`, which checks for coverage, which just isn't a useful metric for us downstream.

Then remove `--cov` arguments;

```
  postPatch = ''
    sed -i "/--cov/d" setup.cfg
  '';

mcwitt replied:
Thanks! There are no `nativeBuildInputs` here, so I was thinking to move this before `propagatedBuildInputs` (where `nativeBuildInputs` would be if it existed). Or do you mean to put it after `nativeCheckInputs` in this case?

mweinelt replied:
No, your first assumption is correct.

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,58 @@
+  pytestFlagsArray = [ "test/unit" ];
+
+  meta = {
+    description = "Redshift interface library";
```
```suggestion
    changelog = "https://github.com/aws/amazon-redshift-python-driver/releases/tag/v${version}";
    description = "Redshift interface library";
```

Reference the changelog URL pattern.

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,58 @@
+    description = "Redshift interface library";
+    homepage = "https://github.com/aws/amazon-redshift-python-driver";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.mcwitt ];
```
```suggestion
    maintainers = with lib.maintainers; [ mcwitt ];
```

More ergonomic and future-proof for adding/removing maintainers.

mcwitt commented:
Thanks for the review @mweinelt! I think I've addressed your comments, except maybe for the ordering of `postPatch` (see https://github.com/NixOS/nixpkgs/pull/213053#discussion_r1089800881)

mweinelt commented on pkgs/development/python-modules/redshift-connector/default.nix:
```diff
@@ -0,0 +1,60 @@
+  ];
+
+  # integration tests require a Redshift cluster
+  pytestFlagsArray = [ "test/unit" ];
```
And finally the following to fix tests on darwin:

```suggestion
  pytestFlagsArray = [ "test/unit" ];
  
  __darwinAllowLocalNetworking = true;
```

````pytb
redshift-connector-aarch64-darwin> =================================== FAILURES ===================================
redshift-connector-aarch64-darwin> __________________ test_get_listen_socket_chooses_free_socket __________________
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin>     def test_get_listen_socket_chooses_free_socket():
redshift-connector-aarch64-darwin>         bacp: BrowserAzureCredentialsProvider = make_valid_browser_azure_credential_provider()
redshift-connector-aarch64-darwin>         ports: typing.Set[int] = set()
redshift-connector-aarch64-darwin>         sockets: typing.List["socket.socket"] = []
redshift-connector-aarch64-darwin>     
redshift-connector-aarch64-darwin>         for _ in range(5):  # open up 10 listen sockets
redshift-connector-aarch64-darwin> >           _socket: "socket.socket" = bacp.get_listen_socket()
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> test/unit/plugin/test_browser_azure_credentials_provider.py:129: 
redshift-connector-aarch64-darwin> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> self = <redshift_connector.plugin.browser_azure_credentials_provider.BrowserAzureCredentialsProvider object at 0x1038bb460>
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin>     def get_listen_socket(self: "BrowserAzureCredentialsProvider") -> socket.socket:
redshift-connector-aarch64-darwin>         s: socket.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
redshift-connector-aarch64-darwin> >       s.bind(("127.0.0.1", 0))  # bind to any free port
redshift-connector-aarch64-darwin> E       PermissionError: [Errno 1] Operation not permitted
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> redshift_connector/plugin/browser_azure_credentials_provider.py:37: PermissionError

mcwitt replied:
ahh thank you, that would've taken me a long time to figure out. Added.

mweinelt commented:
@ofborg build python310.pkgs.redshift-connector
@ofborg build python311.pkgs.redshift-connector
